src/components/Reports/GoalCalculator/Shared/GoalCalculatorContext.tsx

```diff
@@ -112,6 +122,55 @@ export const GoalCalculatorProvider: React.FC<Props> = ({ children }) => {
     setIsDrawerOpen((prev) => !prev);
   }, []);
 
+  const getFamilySections = useCallback(
+    (budgetFamily: BudgetFamilyFragment): SectionItem[] => {
+      return (
+        budgetFamily.primaryBudgetCategories.map((category) => ({
+          title: category.label,
+          complete:
+            category.directInput !== null ||
+            category.subBudgetCategories.some(
+              (category) => category.amount > 0,
+            ),
+        })) ?? []
+      );
+    },
+    [],
+  );
+
+  const percentComplete = useMemo(() => {
```

Comment:
Could the logic to calculate the percentage be pulled out of the context into a helper method? That would make it easier to test. It's fine to keep the `percentComplete` value in context, just move the logic to a helper and call it here something like `const percentComplete = useMemo(() => calculatePercentComplete(goalCalculationResult.data?.goalCalculation), [goalCalculationResult.data])`.